### ISSUE TITLE
Empty docblocks are added above functions on lint

### ISSUE THREAD

**Issue by ay13** [gajus/eslint-plugin-jsdoc#372]
When I run the linter at adds empty docblocks above every function that doesn't already have a docblock. I assume this is part of the auto fixing but this messes doc coverage reports by incorrectly reporting that a function is documented when it is not. 

Is there a way to disable the behavior of auto adding empty doc blocks?

**Comment by brettz9** [comment]
ESLint unfortunately does not provide a means of selectively disabling rules for fixing; one apparently needs to specify a separate config for fixing (where the fixing config, in this case, would omit/disable the `require-jsdoc` rule).

@gajus and @golopot, what do you think of a `disableFixer` option that could be added to each rule to disable the fixer for any rule with a fixer? #336 already anticipates separate options for those rules that have multiple possible fixing strategies, so it might not be too big of a change.

It'd be nice if eslint offered an easier way, e.g., while loading a config, be able to disable specific rules within that config dynamically on the command line, but I received no feedback, at least on this comment on a related issue: https://github.com/eslint/eslint/issues/11747#issuecomment-516586285 .

**Comment by gajus** [comment]
I am probably misunderstanding something. If the intention is to have jsdoc block for every function, then `require-jsdoc` should be enabled and it is expected to "fix" codebase. If there is no such intention, then `require-jsdoc` should not be enabled and it is not anticipated to "fix" codebase.

To me this particular issue sounds like it is more about strictness of `require-jsdoc`. I would argue that a fixer that simply creates an empty jsdoc block doesn't add much value and just creates noise.

**Comment by brettz9** [comment]
I am seeing the issue as the fact that projects will probably not want to go to the trouble of dynamically generating their config files (so that they could, for example, enable `require-jsdoc` in one non-fixing config to get linting warnings where jsdoc blocks were missing, but have the rule disabled in a separate `--fix` config so as not to get the comment block added), and would rather just use a single config file for both identifying linting issues and fixing them.

A related problem as I see it is that projects may differ in what they consider to be a reasonable fix. I would personally find the automatic addition of comment blocks to be useful, especially if starting with a project which had many items to be commented--it would also serve as a visual reminder of the need to document and an invitation to do so more quickly.

As discussed in #47, there may be other fixers which might be desired by some projects, but not others; for example, stripping bad param names might be seen as useful for removing stale param info by some projects, or a risk for removing mere bad spellings by another.

**Comment by carlrannaberg** [comment]
This rule being as fixable is not very useful for us. Especially if it tries to fix it even when it is reported as warning.
This is what happens when this rule is turned on:
Before:
```javascript
module.exports = {
    set(key, value, ttl, callback) {
                // code
        }
```
After:
```
set/**
        *
        *//**
           *
           *//**
              *
              *//**
                 *
                 *//**
 *
 */
        (key, value, ttl, callback) {
                // code
        }
```
Having this kind of result unfortunately makes this rule pretty much useless for us. It would be great if the autofix could be turned off for this rule as I only want to see warnings about functions without JSDoc comments. Any idea how to accomplish that?

**Comment by brettz9** [comment]
@carlrannaberg : I think this might related to problems in the stringifier of `comment-parser` (or a limitation of what the stringifier could represent). I'm afraid I'm a bit tied up to investigate this. As a workaround, I can only suggest having a separate config, one which is for checking only (and has the rule enabled) and one for fixing (which has it disabled). But I'd suggest filing that as a separate issue as this particular issue is for an enhancement to disable fixing, but your case looks like a problem with the fixing itself.

**Comment by brettz9** [comment]
Btw, @carlrannaberg, not sure if the issue was fixed by #524, but I am no longer seeing the recursive issue mentioned in https://github.com/gajus/eslint-plugin-jsdoc/issues/372#issuecomment-591945992

**Comment by brettz9** [comment]
@gajus , @golopot : I really hope we can at least keep the addition of empty jsdoc blocks as an option, as I find them very useful not only for saving a little time in data input but also for being a visual reminder of the need to document (and if one doesn't want empty descriptions to get past linting, one can use `match-description` to report on this). And getting documentation added is also useful because with rules being atomic, rules like `require-returns`, etc. won't report if the jsdoc block is missing, so allowing the auto-addition of doc blocks exposes a fuller list of required documentation to-dos.

However, as another request has been made to disable such a fixer (#594 ) I can make a PR to avoid the fixer, adding a `enableFixer` boolean (such as we have with the `require-param` and `check-param-names` rules) if that is all right. Let me know if you want a somewhat breaking change to make the option (to enable the fixer) off by default.

**Comment by brettz9** [comment]
Note that the new `enableFixer` option is on by default currently, so to avoid the fixer, you have to explicitly set it to `false`.

**Comment by michaelfarrell76** [comment]
hey @brettz9 any chance you have an example with the `match-description` setup? love that it generates the empty comment blocks but just trying to get the linter to report those empty comment blocks as errors

**Comment by brettz9** [comment]
You shouldn't use `valid-jsdoc` as that is ESLint's own rule which is now deprecated. See the alternatives at https://github.com/gajus/eslint-plugin-jsdoc/wiki/Comparison-with-deprecated-JSdoc-related-ESLint-rules .

For `match-description`, you should be able to add (for the most minimal required description) something like:

```js
'jsdoc/match-description': ['error', {
    mainDescription: '.+',
    contexts: ['TSPropertySignature']
}]
```

**Comment by michaelfarrell76** [comment]
thank you so much!!! you're amazing. im getting some weird behavior though where the regex seems to be validating on everything except the empty string

I tried your example at first and wasn't seeing any errors

then i tried this test example to make sure it was reporting and weirdly it reported on all type properties except the ones that were empty 😕 

<img width="388" alt="Screen Shot 2020-09-02 at 8 10 57 PM" src="https://user-images.githubusercontent.com/10264973/92067164-80153580-ed58-11ea-87bf-e0ea8539f70d.png">



<img width="373" alt="Screen Shot 2020-09-02 at 8 10 51 PM" src="https://user-images.githubusercontent.com/10264973/92067160-7ee40880-ed58-11ea-9807-fa211754da59.png">

**Comment by brettz9** [comment]
Whoops, sorry! I forgot that we use `jsdoc/require-description` for the case of an entirely missing description. So you can instead use:

```js

'jsdoc/require-description': ['error', {
    contexts: ['TSPropertySignature']
}]
```

**Comment by michaelfarrell76** [comment]
you are my life saver. appreciate your time!!

**Comment by brettz9** [comment]
My pleasure!
